### PR TITLE
[Merged by Bors] - chore: backports for leanprover/lean4#4814 (part 32)

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
Could you rename the sections per https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/New.20variable.20inclusion.20mechanism/near/457064023?

File: Mathlib/CategoryTheory/Triangulated/HomologicalFunctor.lean [hunk 1]
```diff
@@ -52,16 +52,20 @@ namespace CategoryTheory
...
   [Category D] [HasZeroObject D] [HasShift D ℤ] [Preadditive D]
   [∀ (n : ℤ), (CategoryTheory.shiftFunctor D n).Additive] [Pretriangulated D]
-  [Category A] [Abelian A]
+  [Category A]
 
 namespace Functor
 
 variable (F : C ⥤ A)
 
+section CCACA
```
Comment (sgouezel) [inline Mathlib/CategoryTheory/Triangulated/HomologicalFunctor.lean:64]: Please rename the section

File: Mathlib/Geometry/Manifold/ContMDiff/Defs.lean [hunk 2]
```diff
@@ -342,6 +340,58 @@ theorem smoothAt_iff_target {x : M} :
     SmoothAt I I' f x ↔ ContinuousAt f x ∧ SmoothAt I 𝓘(𝕜, E') (extChartAt I' (f x) ∘ f) x :=
   contMDiffAt_iff_target
 
+section SMWC
```
Comment (sgouezel) [inline Mathlib/Geometry/Manifold/ContMDiff/Defs.lean:343]: rename the section

File: Mathlib/Geometry/Manifold/ContMDiff/Defs.lean [hunk 3]
```diff
@@ -56,12 +56,10 @@ variable {𝕜 : Type*} [NontriviallyNormedField 𝕜]
   {E : Type*}
   [NormedAddCommGroup E] [NormedSpace 𝕜 E] {H : Type*} [TopologicalSpace H]
   (I : ModelWithCorners 𝕜 E H) {M : Type*} [TopologicalSpace M] [ChartedSpace H M]
-  [SmoothManifoldWithCorners I M]
   -- declare a smooth manifold `M'` over the pair `(E', H')`.
```
Comment (sgouezel) [inline Mathlib/Geometry/Manifold/ContMDiff/Defs.lean:59]: The comment does not really make sense any more. Maybe
```suggestion
  -- prerequisite typeclasses to say that `M'` is a smooth manifold over the pair `(E', H')`.
```
Same thing a few lines above and a few lines below

File: Mathlib/RingTheory/Generators.lean [hunk 4]
```diff
@@ -490,6 +504,10 @@ lemma Cotangent.map_id :
   simp only [map_mk, Hom.toAlgHom_id, AlgHom.coe_id, id_eq, Subtype.coe_eta, val_mk,
     LinearMap.id_coe]
 
+variable [Algebra R R''] [IsScalarTower R R' R''] [IsScalarTower R' R'' S'']
+  [IsScalarTower R' S' S''] [IsScalarTower S S' S'']
+variable [Algebra R S''] [IsScalarTower R R'' S''] [IsScalarTower R S S'']
```
Comment (sgouezel) [inline Mathlib/RingTheory/Generators.lean:509]:
```suggestion
  [Algebra R S''] [IsScalarTower R R'' S''] [IsScalarTower R S S'']
```

Comment (Parcly-Taxel) [reply Mathlib/RingTheory/Generators.lean]: All done.

File: Mathlib/Algebra/DirectLimit.lean [hunk 5]
```diff
@@ -269,7 +271,7 @@ theorem totalize_of_not_le {i j} (h : ¬i ≤ j) : totalize G f i j = 0 :=
 
 end Totalize
 
-variable [DirectedSystem G fun i j h => f i j h]
+variable [DecidableEq ι] [DirectedSystem G fun i j h => f i j h] {G f}
```
Comment (semorrison) [inline Mathlib/Algebra/DirectLimit.lean:274]:
```suggestion
variable [DecidableEq ι] [DirectedSystem G fun i j h => f i j h] 
variable {G f}
```
We should always separate binder updates from new variables. Eventually I'd like a linter enforcing this.

**Comment by semorrison** [comment]
bors d+

**Pull request closed without merging**